plugin/trino-hive/src/main/java/io/trino/plugin/hive/InternalHiveConnectorFactory.java

```diff
@@ -127,7 +128,7 @@ public static Connector createConnector(
                     new HiveProcedureModule(),
                     new MBeanServerModule(),
                     binder -> {
-                        binder.bind(OpenTelemetry.class).toInstance(context.getOpenTelemetry());
+                        binder.bind(OpenTelemetry.class).toInstance(openTelemetry.orElse(context.getOpenTelemetry()));
```

Comment:
nice!

Maybe we could replace `TestIcebergGlueCatalogAccessOperations.StealStatsModule` with similar approach.